uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/entitlement-model.js

```diff
@@ -0,0 +1,73 @@
+define([
+  'leave-absences/shared/modules/models',
+  'common/models/model'
+], function (models) {
+  'use strict';
+
+  models.factory('Entitlement', [
+    '$log', 'Model', 'EntitlementAPI', 'EntitlementInstance',
+    function ($log, Model, entitlementAPI, instance) {
+
+      return Model.extend({
+        /**
+         * Calls the all() method of the Entitlement API, and returns an
+         * EntitlementInstance for each entitlement.
+         * It can pass the withBalance property to get remainder for entitlements.
+         *
+         * @param {Object} params matches the api endpoint params (period_id, contact_id, etc)
+         * @param {boolean} withBalance to return remainder of data
+         * @return {Promise}
+         */
+        all: function (params, withBalance) {
+
+          return entitlementAPI.all(params, withBalance)
+            .then(function (entitlements) {
+
+              return entitlements.map(function (entitlement) {
+
+                return instance.init(entitlement, true);
+              });
+            });
+        },
+        /**
+         * Calls the `breakdown` method of the entitlement API
+         *
+         * The return value of the promise changes based on whether an array of `EntitlementInstance`s
+         * has been passed to the method or not.
+         *
+         * If it hasn't, then it simply return the API response.
+         * If it has, then it loads the breakdown data into each corrispondent entitlement, which then
+         * are returned back.
+         *
+         * @param {Object} params matches the api endpoint params (period_id, contact_id, etc)
+         * @param {Array}  [entitlements] an array of `EntitlementInstance`s
+         * @return {Promise}
+         */
+        breakdown: function (params, entitlements) {
+          return entitlementAPI.breakdown(params)
+            .then(function (breakdown) {
+              if (entitlements) {
+                entitlements.map(function (entitlement) {
+                  var foundEntitlement = _.find(breakdown, function (element) {
+                    return element.id === entitlement.id;
+                  });
+                  if (foundEntitlement) {
+                    entitlement['breakdown'] = foundEntitlement['breakdown'];
+                  }
+
+                  return entitlement;
+                });
+
+                return entitlements;
+              }
+
+              return breakdown.map(function (entitlement) {
```

Comment:
changed documentation to 
`
* If it hasn't, then it returns the entitlements along with breakdown details based on params passed.
`

essentially its similar to all call but returning entitlements instance with breakdown data if existing entitlements not passed